File: app/src/main/java/com/supercilex/robotscouter/data/client/UploadTeamMediaJob14.java
package com.supercilex.robotscouter.data.client;

import com.firebase.jobdispatcher.JobParameters;
import com.firebase.jobdispatcher.JobService;
import com.supercilex.robotscouter.data.remote.TbaUploader;
import com.supercilex.robotscouter.data.util.TeamHelper;
import com.supercilex.robotscouter.util.JobUtils;

public class UploadTeamMediaJob14 extends JobService {
    @Override
    public boolean onStartJob(JobParameters params) {
        TeamHelper oldTeamHelper = JobUtils.parseRawBundle(params.getExtras());
        TbaUploader.upload(oldTeamHelper.getTeam(), this)
                .addOnSuccessListener(newTeam -> {
                    oldTeamHelper.updateMedia(newTeam);
                    jobFinished(params, false);
                })
                .addOnFailureListener(e -> jobFinished(params, true));
        return true;
    }

    @Override
    public boolean onStopJob(JobParameters job) {
        return true;
    }
}

File: app/src/main/java/com/supercilex/robotscouter/data/client/UploadTeamMediaJob14.kt
package com.supercilex.robotscouter.data.client

import com.google.android.gms.tasks.Task
import com.supercilex.robotscouter.data.model.Team
import com.supercilex.robotscouter.data.remote.TbaUploader
import com.supercilex.robotscouter.util.updateMedia

class UploadTeamMediaJob14 : TbaJobBase14() {
    override val updateTeam: (team: Team, newTeam: Team) -> Unit
        get() = { team, newTeam -> team.updateMedia(newTeam) }

    override fun startTask(previousTeam: Team): Task<Team> = TbaUploader.upload(previousTeam, this)
}
